python/common/org/Python.java

```diff
@@ -602,12 +602,45 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
                     "iteration.  The enumerate object yields pairs containing a count (from\n" +
                     "start, which defaults to zero) and a value yielded by the iterable argument.\n" +
                     "enumerate is useful for obtaining an indexed list:\n" +
-                    "       (0, seq[0]), (1, seq[1]), (2, seq[2]), ...\n"
+                    "       (0, seq[0]), (1, seq[1]), (2, seq[2]), ...\n",
+           args = {"items"},
+           default_args = {"start"}
     )
-    public static org.python.Iterable enumerate() {
-        throw new org.python.exceptions.NotImplementedError("Builtin function 'enumerate' not implemented");
+    //public static org.python.Iterable enumerate(org.python.Iterable items, org.python.types.Int start) {
+   public static org.python.Iterable enumerate(org.python.Object items,/*org.python.types.List items,*/ org.python.Object start) {
+    	try{
+    		if(start==null)
+    			start=new org.python.types.Int(0);
+    		else {
+    			if(!(start instanceof org.python.types.Int))
+    				throw new org.python.exceptions.TypeError("integer argument expected as 'start' , got " + start.typeName() + "");
```

Comment:
Note that these errors should match the errors in CPython, this looks different than what I get on Python 3.4:

```
>>> list(enumerate([], start=None))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object cannot be interpreted as an integer
```

Can you add some tests that exercise this code?

Something like:

```
def test_enumerate_invalid_start_args(self):
       self.assertCodeExecution("""
            print(list(enumerate(['a','b','c'], start=None)))
            """, exits_early=True)
       self.assertCodeExecution("""
            print(list(enumerate(['a','b','c'], start=1.5)))
            """, exits_early=True)
       self.assertCodeExecution("""
            print(list(enumerate(['a','b','c'], start=None)))
            """, exits_early=True)
```

Using `instanceof` here is probably not be the best, try adding a test for passing a boolean:

    print(list(enumerate(['a','b','c'], start=True))) # in Python, True has integer value of 1

You may try simply casting for now and then throw the proper error if casting fails.